### PR TITLE
Knex

### PR DESCRIPTION
**Pull request by runeh**
Tried using knex for migrations, ended up using it for everything else as well.. . Dunno if that's what you want?

Needs readme updates etc. `knex migrate:latest` in the root to initialize db.

### CODE REVIEW

**Comment by runeh** [comment]
Oh, and it broke timestamp stuff. It's sending iso strings. Can switch it to timestamps. I also thing it updates `created_at` now whenever the row is thouched, which is obviously wrong.

**Review by theneva — COMMENTED** [review]
This definitely looks like the right way to go, very nice! :smile:

File: server/index.js [hunk 1]
```diff
@@ -1,109 +1,109 @@
 const express = require('express');
-const pg = require('pg');
+const knex = require('knex');
+const dbConfig = require('../knexfile');
 
-const actualSubmit = pg.Query.prototype.submit;
-
-pg.Query.prototype.submit = function() {
-  console.log(this.text);
```
[the remark this answers is not in the document]
Comment (runeh) [reply server/index.js]: Yup. It does it here https://github.com/rantlr/monorepo/pull/18/files#diff-0861d6d6b50d7d695344bf2d86d6e5e6R11

[the remark this answers is not in the document]
Comment (runeh) [reply server/index.js]: THere are some other possible events there that might be useful I think

File: scripts/run-pg-docker.sh [hunk 2]
```diff
@@ -7,6 +7,9 @@ echo "Creating network $network_name..."
 docker network create $network_name
 echo "Created network $network_name"
 
+echo "Creating /tmp/rants-data for database volume"
+mkdirp /tmp/rants-data
```
Comment (theneva) [inline scripts/run-pg-docker.sh]: this should probably be `mkdir -p /tmp/rants-data`. I don't have `mkdirp` installed globally :smile:

**Comment by theneva** [comment]
Am I right to think that we can just delete the `wipe-db` and `init-db` scripts too, since you can just do `knex migrate:rollback` instead of running the scripts?

If true, that means we no longer need to connect to the Postgres Docker container from other containers—so we can drop the `--name` and `--network` stuff, and don't need to trap `sigint` anymore.

This removes a lot of complexity and weird hacks :heart:

File: scripts/run-pg-docker.sh (hunk 2 above)
Comment (runeh) [reply scripts/run-pg-docker.sh]: Heh. Yes. That's what I meant.

**Comment by runeh** [comment]
I deleted the initdb script. Not sure how to migrate down from the initial migration. Will check. Worst case, we can do `rm -rf /tmp/rants-data`

**Comment by theneva** [comment]
I'm fine with removing /tmp/rants-data, and (if it's not too much work) killing and restarting the postgres container script if it's running!

**Comment by theneva** [comment]
@runeh it seems like knex doesn't do anything to maintain either timestamp on updates. How do you feel about manually creating the timestamps (`table.timestamp('created').defaultTo(knex.fn.now())` and same for `updated`) to prevent linters from yelling at snake_cased names, since we have to care about them anyway?

**Comment by theneva** [comment]
After timestamps work I'm fine with merging this :) no need to fix all the script stuff in one go

**Pull request merged**